### PR TITLE
Test 1.0

### PR DESCRIPTION
**Pull request by aksoli666**
controller-test-coverage
service-test-coverage
added tests for Book repository and tests for service, controller layers for Book and Category

### CODE REVIEW

File: src/test/java/shop/controller/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+import org.springframework.web.context.WebApplicationContext;
+import shop.dto.request.category.CreateCategoryRequestDto;
+import shop.dto.request.category.UpdateCategoryRequestDto;
+import shop.dto.responce.category.CategoryDto;
+
+import static org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder.reflectionEquals;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CategoryControllerTest {
+    private static final String ADD_CATEGORIES_SQL =
```
Comment (Elena-Bruyako) [inline src/test/java/shop/controller/CategoryControllerTest.java]: it's better to remove these constants and use `@Sql` under the test class with all scripts that you need

File: src/test/java/shop/service/BookServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    @Test
+    @DisplayName("""
+            Verify getBookById(...), return BookDto
+            """)
+    public void getBookById_ValidId_returnBookDto() {
+        Category category = new Category();
+        category.setName(CATEGORY_NAME_CORRECT);
+
+        Book book = new Book();
+        book.setId(ID_1L_CORRECT);
```
Comment (Elena-Bruyako) [inline src/test/java/shop/service/BookServiceTest.java]: check if you have duplication and use separate methods for creating test data 
Also, you can create TestUtil class with methods that return test data and TestConstants for constants

File: src/test/resources/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+databaseChangeLog:
```
Comment (Elena-Bruyako) [inline src/test/resources/changelog/db.changelog-master.yaml:1]: redundant

File: src/test/resources/database/shop/category/delete-categories.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+DELETE FROM categories WHERE id IN (6, 7);
```
Comment (Elena-Bruyako) [inline src/test/resources/database/shop/category/delete-categories.sql:1]: why do you need so many delete scripts?

File: src/test/resources/database/shop/category/add-category-for-delete-by-id.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+INSERT INTO categories (id, name, description, is_deleted) VALUES (10, "Y Category", "Y Description for Category Y", 0)
```
Comment (Elena-Bruyako) [inline src/test/resources/database/shop/category/add-category-for-delete-by-id.sql]: why do you need so many insert scripts?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**